File: hackerrank/divisiblesums.hs
-- @Author: krocki
-- @Date:   2017-01-04 16:01:52
-- @Last Modified by:   krocki
-- @Last Modified time: 2017-01-04 16:02:22

count_sums xs k = sum [1| i <- [0..(length xs-1)], j <- [0..(length xs-1)], (xs !! i + xs !! j) `mod` k == 0, i > j ]


main :: IO ()
main = do
    n_temp <- getLine
    let n_t = words n_temp
    let n = read $ n_t!!0 :: Int
    let k = read $ n_t!!1 :: Int
    a_temp <- getLine
    let a = map read $ words a_temp :: [Int]
        sums = count_sums a k
    print sums

File: hackerrank/divisiblesums.py
# -*- coding: utf-8 -*-
# @Author: krocki
# @Date:   2017-01-04 15:46:13
# @Last Modified by:   krocki
# @Last Modified time: 2017-01-04 15:46:25

#!/bin/python

import sys


n,k = raw_input().strip().split(' ')
n,k = [int(n),int(k)]
a = map(int,raw_input().strip().split(' '))
counter = 0

for i in range(0, n):
    for j in range(i+1, n):
        s = a[i] + a[j]
        if s % k == 0: counter = counter+1

print counter